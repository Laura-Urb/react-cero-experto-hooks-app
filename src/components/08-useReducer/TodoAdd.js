import React from 'react';
import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({ handleAddTodo }) => {
  const [{ descripcion }, handleInputChange, reset] = useForm({
    descripcion: '',
  });
  const handleSubmit = (e) => {
    e.preventDefault();
    if (descripcion.trim().length <= 1) return;

    const newTodo = {
      id: new Date().getTime().toString(),
      desc: descripcion,
      done: false,
    };

    handleAddTodo(newTodo);
    reset();
  };
  return (
    <>
      <h4>Agregar TODO</h4>
      <hr />
      <form onSubmit={handleSubmit}>
        <input
          type=''
          name='descripcion'
          value={descripcion}
          placeholder='Aprender...'
          autoComplete='off'
          className='form-control'
          onChange={handleInputChange}
        />
        <button type='submit' className='btn btn-outline-primary mt-1 w-100'>
          Agregar
        </button>
      </form>
    </>
  );
};
