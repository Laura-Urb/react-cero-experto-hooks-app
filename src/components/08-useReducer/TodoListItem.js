import React from 'react';

export const TodoListItem = ({ todo, i, handleDelete, handleToogle }) => {
  return (
    <li className='list-group-item'>
      <p
        className={`${todo.done && 'complete'}`}
        onClick={() => handleToogle(todo.id)}
      >
        {i + 1}. {todo.desc}
      </p>
      <button onClick={() => handleDelete(todo.id)} className='btn btn-danger'>
        Borrar
      </button>
    </li>
  );
};
