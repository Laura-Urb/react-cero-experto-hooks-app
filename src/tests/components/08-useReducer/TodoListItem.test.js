import { shallow } from 'enzyme';
import { TodoListItem } from '../../../components/08-useReducer/TodoListItem';
import React from 'react';
import { demoTodos } from '../../fixtures/demoTodos';
describe('Pruebas TodoListItem', () => {
  const handleDelete = jest.fn();
  const handleToogle = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      i={0}
      handleDelete={handleDelete}
      handleToogle={handleToogle}
    />
  );
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de llamar la funcion borrar', () => {
    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test('debe de llamar la funcion toggle', () => {
    wrapper.find('p').simulate('click');
    expect(handleToogle).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test('debe de mostrarse el texto correctamente', () => {
    const p = wrapper.find('p').text();
    expect(p).toBe('1. ' + demoTodos[0].desc);
  });
  test('debe de tener la clase complete si el todo.done estÃ¡ en true', () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        i={0}
        handleDelete={handleDelete}
        handleToogle={handleToogle}
      />
    );
    expect(wrapper.find('p').hasClass('complete')).toBe(true);
  });
});
