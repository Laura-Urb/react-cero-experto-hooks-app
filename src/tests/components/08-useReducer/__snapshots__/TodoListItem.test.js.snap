// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas TodoListItem debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    AAA
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
